modules/core/src/api.ts

```diff
@@ -0,0 +1,127 @@
+      serialize = superagent.serialize['application/json'];
+    }
+    if (serialize) {
+      data = serialize(data);
+      (req as any)._data = data;
+      return data;
+    }
+  }
+}
+
+/**
+ * Set the superagent query string correctly for browsers or node.
+ * @param req
+ */
+export function setRequestQueryString(req: superagent.SuperAgentRequest): void {
+  const urlDetails = urlLib.parse(req.url);
+
+  let queryString: string | undefined;
+  const query: string[] = (req as any)._query;
+  const qs: { [key: string]: string } = (req as any).qs;
+  if (query && query.length > 0) {
+    // browser version
+    queryString = query.join('&');
+    (req as any)._query = [];
+  } else if (qs) {
+    // node version
+    queryString = querystring.stringify(qs);
+    (req as any).qs = null;
+  }
+
+  if (queryString) {
+    if (urlDetails.search) {
+      urlDetails.search += '&' + queryString;
+    } else {
+      urlDetails.search = '?' + queryString;
+    }
+    req.url = urlLib.format(urlDetails);
+  }
+}
+
+/**
+ * Verify that the response received from the server is signed correctly.
+ * Right now, it is very permissive with the timestamp variance.
+ */
+export function verifyResponse(
+  bitgo: BitGo,
+  token: string | undefined,
+  method: VerifyResponseOptions['method'],
+  req: superagent.SuperAgentRequest,
+  response: superagent.Response
+): superagent.Response {
+  // we can't verify the response if we're not authenticated
+  if (!req.isV2Authenticated || !req.authenticationToken) {
+    return response;
+  }
+
+  const verificationResponse = bitgo.verifyResponse({
+    url: req.url,
+    hmac: response.header.hmac,
+    statusCode: response.status,
+    text: response.text,
+    timestamp: response.header.timestamp,
+    token: req.authenticationToken,
+    method,
+  });
+
+  if (!verificationResponse.isValid) {
+    // calculate the HMAC
+    const receivedHmac = response.header.hmac;
+    const expectedHmac = verificationResponse.expectedHmac;
+    const signatureSubject = verificationResponse.signatureSubject;
+    // Log only the first 10 characters of the token to ensure the full token isn't logged.
+    const partialBitgoToken = token ? token.substring(0, 10) : '';
+    const errorDetails = {
+      expectedHmac,
+      receivedHmac,
+      hmacInput: signatureSubject,
+      requestToken: req.authenticationToken,
+      bitgoToken: partialBitgoToken,
+    };
+    debug('Invalid response HMAC: %O', errorDetails);
+    throw new ApiResponseError('invalid response HMAC, possible man-in-the-middle-attack', 511, errorDetails);
+  }
+
+  if (bitgo.getAuthVersion() === 3 && !verificationResponse.isInResponseValidityWindow) {
+    const errorDetails = {
+      timestamp: response.header.timestamp,
+      verificationTime: verificationResponse.verificationTime,
+    };
+    debug('Server response outside response validity time window: %O', errorDetails);
+    const error: any = new Error(
```

Comment:
Why are you using a regular `Error` object here cast to `any` instead of the `ApiResponseError` like a few lines above? It looks like that class has the `status` and `result` fields built in.

Edit: I realize now this is old code that's being moved from one file to another, but I guess the question still stands if only for curiosity's sake.

Reply:
that's a good point, I will do this in a follow up